src/BloomExe/Book/Page.cs

```diff
@@ -154,5 +154,21 @@ public static string GetPageSelectorXPath(XmlDocument pageDom)
 			var id = pageDom.SelectSingleNode("/html/body/div").Attributes["id"].Value;
 			return string.Format("/html/body/div[@id='{0}']", id);
 		}
+
+		/// <summary>
+		/// Return the index of this page in the IEnumerable of pages
+		/// </summary>
+		/// <returns>Index of the page, or -1 if the page was not found</returns>
+		public int GetIndex()
+		{
+			var i = 0;
+			foreach (var page in Book.GetPages())
+			{
+				if (page.Id == Id) return i;
```

Comment:
I'm surprised the test isn't, if (page == this). Do we have multiple page objects representing the same page?